File: frontend/src/api/ocr_config.ts
import client from './client'

export interface OCRConfig {
  id: number
  name: string
  provider: string
  api_base_url: string | null
  language: string
  is_active: boolean
  description: string | null
  created_at: string | null
}

export interface OCRConfigForm {
  name: string
  provider?: string
  api_base_url?: string
  api_key?: string
  language?: string
  description?: string
}

const base = '/api/ocr-configs'

export const ocrConfigApi = {
  list() { return client.get(base) },
  create(form: OCRConfigForm) { return client.post(base, form) },
  update(id: number, form: Partial<OCRConfigForm & { is_active: boolean }>) {
    return client.put(`${base}/${id}`, form)
  },
  delete(id: number) { return client.delete(`${base}/${id}`) },
}

File: backend/app/models/ocr_config.py
"""OCR model configuration — API-based OCR."""

from datetime import datetime, timezone
from sqlalchemy import String, Integer, Boolean, Text, DateTime, ForeignKey, func
from sqlalchemy.orm import Mapped, mapped_column, relationship
from app.models.user import Base


class OCRConfig(Base):
    __tablename__ = "ocr_configs"

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
    user_id: Mapped[int] = mapped_column(ForeignKey("users.id", ondelete="CASCADE"), index=True)
    name: Mapped[str] = mapped_column(String(200), nullable=False)
    provider: Mapped[str] = mapped_column(String(50), default="custom")  # custom API
    api_base_url: Mapped[str | None] = mapped_column(String(500), nullable=True)
    api_key: Mapped[str | None] = mapped_column(Text, nullable=True)
    language: Mapped[str] = mapped_column(String(50), default="ch")
    is_active: Mapped[bool] = mapped_column(Boolean, default=False)
    description: Mapped[str | None] = mapped_column(Text, nullable=True)
    created_at: Mapped[datetime] = mapped_column(
        DateTime, default=lambda: datetime.now(timezone.utc), server_default=func.now()
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime,
        default=lambda: datetime.now(timezone.utc),
        onupdate=lambda: datetime.now(timezone.utc),
        server_default=func.now(),
        server_onupdate=func.now(),
    )

    owner: Mapped["User"] = relationship("User", back_populates="ocr_configs")
